src/flamegraph/mod.rs

```diff
@@ -52,6 +52,67 @@ macro_rules! args {
     }};
 }
 
+struct FrameAttributes<'a> {
```

Comment:
Why have a type here that's separate from `FrameAttrs`?

Reply:
`FrameAttrs` owns its fields. We only need to have reference to the fields when drawing the frame. I didn't want to copy some `String` and `Vec`.

Reply:
Wouldn't `Cow` let you merge these types then?

Reply:
Other issue: most `FrameAttrs` fields are optional, which would add useless `unwrap` for some values that we know have values (because we have a default value ready to be used)

Reply:
Okay, I guess you've convinced me. It just still makes me sad :p

Reply:
My goal here was to reduce the complexity of `from_sorted_lines` (at least for clippy). Adding a structure may not be ideal, I may try something else (at least to make clippy happy, we may need to open another issue to fix the problem once and for all).